sycl/test/devicelib/math_test.cpp

```diff
@@ -4,69 +4,217 @@
 #include <CL/sycl.hpp>
 #include <iostream>
 #include <math.h>
+#include "math_utils.hpp"
 
 namespace s = cl::sycl;
 constexpr s::access::mode sycl_read = s::access::mode::read;
 constexpr s::access::mode sycl_write = s::access::mode::write;
 
-class DeviceSin;
+#define TEST_NUM 38
 
-void device_sin_test(s::queue &deviceQueue) {
-  s::range<1> numOfItems{1};
-  float  result_f = -1;
-  {
-    s::buffer<float, 1> buffer1(&result_f, numOfItems);
-    deviceQueue.submit([&](cl::sycl::handler &cgh) {
-      auto res_access1 = buffer1.get_access<sycl_write>(cgh);
-      cgh.single_task<class DeviceSin>([=]() {
-        res_access1[0] = sinf(0);
-      });
-    });
-  }
+float ref[TEST_NUM] = {
+1, 0, 0, 0, 0, 0, 0, 1, 1, 0.5,
+0, 2, 0, 0, 1, 0, 2, 0, 0, 0,
+0, 0, 1, 0, 1, 2, 0, 1, 2, 5,
+0, 0, 0, 0, 0.5, 0.5, NAN, NAN,};
 
-  assert(result_f == 0);
-}
-
-class DeviceCos;
+float refIptr = 1;
 
-void device_cos_test(s::queue &deviceQueue) {
-  s::range<1> numOfItems{1};
-  float  result_f = -1;
+void device_math_test(s::queue &deviceQueue) {
+  s::range<1> numOfItems{TEST_NUM};
+  float result[TEST_NUM] = {-1};
+  // Variable exponent is an integer value to store the exponent in frexp function
+  int exponent = -1;
+  // Variable iptr stores the integral part of float point in modf function
+  float iptr = -1;
+  // Variable quo stores the sign and some bits of x/y in remquo function
+  int quo = -1;
   {
-    s::buffer<float, 1> buffer1(&result_f, numOfItems);
+    s::buffer<float, 1> buffer1(result, numOfItems);
+    s::buffer<int, 1> buffer2(&exponent, s::range<1>{1});
+    s::buffer<float, 1> buffer3(&iptr, s::range<1>{1});
+    s::buffer<int, 1> buffer4(&quo, s::range<1>{1});
     deviceQueue.submit([&](cl::sycl::handler &cgh) {
-      auto res_access1 = buffer1.get_access<sycl_write>(cgh);
-      cgh.single_task<class DeviceCos>([=]() {
-        res_access1[0] = cosf(0);
+      auto res_access = buffer1.template get_access<sycl_write>(cgh);
+      auto exp_access = buffer2.template get_access<sycl_write>(cgh);
+      auto iptr_access = buffer3.template get_access<sycl_write>(cgh);
+      auto quo_access = buffer4.template get_access<sycl_write>(cgh);
+      cgh.single_task<class DeviceMathTest>([=]() {
+        int i = 0;
+        {
+          float a = 0;
+          res_access[i++] = cosf(a);
+        }
+        {
+          float a = 0;
+          res_access[i++] = sinf(a);
+        }
```

Comment:
Note this code is 90% repeats itself.
Only input value and function name are different.
Can use something like:
```suggestion
          res_access[i++] = cosf(0.f);
          res_access[i++] = sinf(0.f);
```
?